src/proj7BittingCerratoCohenEllmer/bantam/lexer/Scanner.java

```diff
@@ -108,11 +80,11 @@ public Scanner(Reader reader, ErrorHandler handler) {
      */
     public Token scan() {
         Stack<Character> spellingStack = new Stack<>();
-        if (skippedLastToken != '\0') {
+        if (skippedLastToken != '\0' && !Character.isWhitespace(skippedLastToken)) {
             spellingStack.push(skippedLastToken);
         }
 
-        do {
+        while (!isCompleteToken(spellingStack)) {
             try {
                 char letter = sourceFile.getNextChar();
                 if (!Character.isWhitespace(letter) || !spellingStack.empty()) {
```

Comment:
I assume it would not be good to add a flag here that finds if a character is illegal ?
To handle:
Unsupported escape characters within a string
( '?', '@',... ) outside of strings and comments